src/filters/arrays.js

```diff
@@ -50,8 +51,9 @@ function matchItems(array1, array2, index1, index2, context) {
   }
   var objectHash = context.objectHash;
   if (!objectHash) {
-    // no way to match objects was provided, try match by position
-    return context.matchByPosition && index1 === index2;
+    var ivalue1 = Immutable.Map(value1);
+    var ivalue2 = Immutable.Map(value2);
+    return Immutable.is(ivalue1, ivalue2);
```

Comment:
I did some speed tests to see what kind of impact the hash would be so on a 1.1MB[1] json file the overhead of the hash is 0.04 seconds[2]. The memory overhead goes from around 31150 KB to 31250 KB.

Of course I care about that difference, but I think there should be more gains speed wise from using Immutable.js since it's well optimized and you don't have the inherent problem of JavaScript objects and arrays.

[1] http://www.usda.gov/data.json
[2] Without just getting one property value:
0.16-0.23 sec Elapsted time
With same json in Immutable.Map and doing calling hashCode:
0.19-0.27 sec Elaptesd time